### DISCUSSION TITLE
Use CommunityToolkit.Maui or CommunityToolkit.Maui.Core?

### DISCUSSION THREAD

**Discussion by neonleo** [CommunityToolkit/Maui#607] [Q&A] [answered]
Just wonder what is CommunityToolkit.Maui.Core, because all document is talking CommunityToolkit.Maui package, but few mention the CommunityToolkit.Maui.Core, just a line on Nuget - 
`Core library for community toolkits using .NET MAUI`

So which one shoud I use if working MAUI with .NET core 6
Thanks

**Comment by VladislavAntonyuk** [comment, marked as answer]
CommunityToolkit.Maui.Core is abstraction. It contains interfaces, handlers and platform controls. It can be used by other developers to implement there own MAUI controls. 
CommunityToolkit.Maui implements Core. 

For example. If you want to have your own DrawingView or Snackbar you can implement interfaces from Core

**Reply by VladislavAntonyuk** [reply]
In most cases you need CommunityToolkit.Maui

**Reply by neonleo** [reply]
Thanks you, nice explanation 💯

**Comment by toddalbers** [comment]
I think that making tools and references for a programing language or platform as unambiguous as possible helps new developers to stick with that programming language or platform rather than go another route.

I think this explanation below given by [VladislavAntonyuk](https://github.com/VladislavAntonyuk) should be added to both the listing and description of CommunityToolkit.Maui.Core within Visual Studio shown in screenshots below.

> Explanation: CommunityToolkit.Maui.Core is abstraction. It contains interfaces, handlers and platform controls. It can be used by other developers to implement their own MAUI controls.  CommunityToolkit.Maui implements Core.  In most cases you will need CommunityToolkit.Maui, not CommunityToolkit.Maui.Core.

![image](https://user-images.githubusercontent.com/52428561/236871749-c46beb50-f381-416d-93a8-fb8a7ddc9271.png)

![image](https://user-images.githubusercontent.com/52428561/236871885-2ac0c576-085d-4ffc-a5c7-26bd0570c1bf.png)
![image](https://user-images.githubusercontent.com/52428561/236871974-20232ddd-c0fb-4009-838e-ba915c1c7751.png)

Note that the Project URL for both above is the same.  Not sure if this is a simple error or not.

It almost seems like Microsoft doesn't know whether it wants to recognize CommunityToolkit.Maui.Core or not, which just makes it more confusing.  They list it in Visual Studio (shown above) with little explanation, but don't have a learn.microsoft page for it.
It might help if the description was also added to the learn.microsoft pages for maui community toolkit:

https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/

And\or maybe Microsoft should have a page for the CommunityToolkit.Maui.Core, which it doesn't appear to have.

https://learn.microsoft.com/en-us/dotnet/communitytoolkit/

https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/core/    (Nothing here)

It probably would help also if the description was added to the nuget readme pages for both of the following instead of only saying:

https://www.nuget.org/packages/CommunityToolkit.Maui.Core#readme-body-tab
https://www.nuget.org/packages/CommunityToolkit.Maui.Core/5.1.0#readme-body-tab

These above only say: 
>  _Core library for community toolkits using .NET MAUI._

It probably would help also if the description was added to the nuget release pages:

https://www.nuget.org/packages/CommunityToolkit.Maui.Core#release-body-tab
https://www.nuget.org/packages/CommunityToolkit.Maui.Core/5.1.0#release-body-tab

These above only say: 
> _See https://github.com/CommunityToolkit/Maui/releases_

**Reply by TheCodeTraveler** [reply]
Hi @toddalbers! I sense you're frustrated by the lack of documentation for `CommunityToolkit.Maui.Core`. The quickest explanation that I can give is that `CommunityToolkit.Maui.Core` is for library creators (eg if someone were to create a `CommunityToolkit.Maui.BlazorWebView` toolkit, they should leverage `CommunityToolkit.Maui.Core`), and for developers who are not using the cross-platform UI controls found in the `Microsoft.Maui.Controls` namespace (eg they are only using `<UseMauiCore>true</UseMauiCore>` in their `csproj`, not `<UseMaui>true</UseMaui>`). 

It is a very rare use-case for developers to add a NuGet dependency to `CommunityToolkit.Maui.Core` and 99% of .NET MAUI devs won't (and shouldn't). 

I agree that the documentation could be better and a few points you've raised are great. I encourage you to help us by [submitting a PR](https://github.com/MicrosoftDocs/CommunityToolkit) to improve what is missing. We are a team of volunteers and do not get paid by Microsoft to create + maintain the .NET MAUI Community Toolkit. 

I've tried to answer most of your questions / concerns below:

> Note that the Project URL for both above is the same. Not sure if this is a simple error or not

Both `CommunityToolkit.Maui` and `CommunityToolkit.Maui.Core` libraries (ie `CommunityToolkit.Maui.csproj` and `CommunityToolkit.Maui.Core.csproj`) live in the same GitHub repository. Using the same Project URL for both NuGet Packages  is not an error. 

> It almost seems like Microsoft doesn't know whether it wants to recognize CommunityToolkit.Maui.Core or not

I'm not sure what you mean by this. The Community Toolkit is created + maintained by community volunteers like myself. All of us core maintainers donate our time on nights + weekends to support this effort. We are not paid by Microsoft for this work.

The documentation for our library lives on https://learn.microsoft.com, however, we are not a Microsoft funded project.

> They list it in Visual Studio (shown above) with little explanation

I don't understand what you mean here. Who is "they" and where do "they list it in Visual Studio"? Could you share an example from a library that demonstrates what you're looking to achieve?

> And\or maybe Microsoft should have a page for the CommunityToolkit.Maui.Core, which it doesn't appear to have

That'd be great if you could open a PR adding docs for `CommunityToolkit.Maui.Core`! https://github.com/MicrosoftDocs/CommunityToolkit

> It probably would help also if the description was added to the nuget readme pages for both of the following instead of only saying: 

Great idea! I'll submit a PR to update the NuGet description

> It probably would help also if the description was added to the nuget release pages

Our release notes live in GitHub. On NuGet, we point to our GitHub Releases page to direct developers to the latest release notes.

**Comment by CliffAgius** [comment]
Closed as answered
